api/app/src/main/kotlin/packit/clients/GithubUserClient.kt

```diff
@@ -38,14 +37,26 @@ class GithubUserClient(private val config: AppConfig, private val githubBuilder:
         return UserPrincipal.create(user, mutableMapOf())
     }
 
-    fun checkGithubOrgMembership()
+    @Throws(PackitAuthenticationException::class)
+    fun checkGithubMembership()
     {
         checkAuthenticated()
 
-        val userOrgs = ghUser!!.allOrganizations.map {org -> org.login}
-        val inAllowedOrg = allowedOrgs.any {allowed -> userOrgs.contains(allowed)}
+        val userOrg = ghUser!!.allOrganizations.firstOrNull { org -> org.login == config.authGithubAPIOrg }
+        var userAllowed = userOrg != null
 
-        if (!inAllowedOrg)
+        val allowedTeam = config.authGithubAPITeam
+        if (userAllowed && allowedTeam.isNotEmpty())
+        {
+            // We've confirmed user is in org, and required team is not empty, so we need to check team membership too
+            val team = userOrg!!.teams[allowedTeam] ?: throw PackitAuthenticationException(
+                "githubConfigTeamNotInOrg",
+                HttpStatus.UNAUTHORIZED
+            )
+            userAllowed = ghUser!!.isMemberOf(team)
+        }
+
+        if (!userAllowed)
```

Comment:
this logic feels very weird to me to follow, like i get what it is trying to do but the `userAllowed` checks are weird, it goes into a condition that relies on `userAllowed` and then it potentially changes the value of `userAllowed` and we do another if statement on it. Here's my go at writing this function, feel free to not do it this way!
```kotlin
checkAuthenticated()

val userOrg = ghUser!!.allOrganizations.firstOrNull { org -> org.login == config.authGithubAPIOrg }
val allowedTeam = config.authGithubAPITeam
val orgContainsTeam = if (allowedTeam.isEmpty()) true else userOrg?.teams?.containsKey(allowedTeam)

if (userOrg == null || !ghUser!!.isMemberOf(userOrg.teams[allowedTeam])) {
    throw PackitAuthenticationException("githubUserRestrictedAccess", HttpStatus.UNAUTHORIZED)
}

if (orgContainsTeam == false) {
    throw PackitAuthenticationException(
        "githubConfigTeamNotInOrg",
        HttpStatus.UNAUTHORIZED
    )
}
```

Reply:
I don't think that will quite work, because it's purporting to check if user `isMemberOf` team which might be null because  the team isn't in the org (so I don't think it would compile) - we could put an extra check for it to fail that `if` if `!orgContainsTeam`, so that it goes on to raise the `gitHubConfigTeamNotInOrg` error - but I think that would be even more confusing, particularly since `orgContainsTeam` is true if there isn't a configured team... 

I feel like the pattern of keeping a running 'ok' flag and doing successive checks which set it to false if there's anything wrong is fairly standard. The logic as is sort of makes sense as it's doing the team check only once it knows it can rely on the user org. But yeah it is a bit weird. I'll add some comments and change `userAllowed` to `userOK`, as I feel like that sounds a bit less final..?!